### DISCUSSION TITLE
Push signed

### DISCUSSION THREAD

**Discussion by lazize** [community/community#23515] [New to GitHub]
<p>Hi,</p>
<p>Does GitHub support signed push?</p>
<p>I am getting this error message:</p>
<pre><code> $ git push --signed=true
 Username for 'https://github.com': user
 Password for 'https://user@github.com':
 fatal: the receiving end does not support --signed push
 fatal: The remote end hung up unexpectedly
 error: failed to push some refs to 'https://github.com/user/repo.git'
</code></pre>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/lazize">@lazize</a>,</p>
<p>GitHub does support GPG signed commits. For more information, please see the GitHub help article, <a href="https://help.github.com/articles/signing-commits-using-gpg/" rel="nofollow noopener">Signing commits using GPG</a>.</p>
<p>Thanks!</p>

**Comment by lazize** [comment]
<p>Hi,</p>
<p>Yes, it supports signed commit, it means the example below works fine.</p>
<p>Please not that only commit is signed.</p>
<pre><code>git commit -S -m "message"
git push
</code></pre>
<p>But I am asking about signed push, not only signed commit.</p>
<p>Note that on example below both, commit and push, are signed and it fails with the error message I posted.</p>
<pre><code>git commit -S -m "message"
git push --signed=true
</code></pre>

**Comment by lee-dohm** [comment, marked as answer]
<p>No, GitHub does not currently support GPG signed pushes, only signed commits.</p>

**Comment by kushaldas** [comment]
<p>Hi from 2021, can we do <code>GPG signed</code> push now? or enable it somehow?</p>

**Comment by potentialuser54** [comment]
<p>Hi from 2022. Just ran into this.</p>
<p><a class="mention" href="/lee-dohm">@lee-dohm</a> Could you give any reason as to why GitHub does not support signed pushes. and is there any plan to support it?</p>

**Comment by rahilarious** [comment]
<p>it seems that neither github or gitlab supports it. :pensive:</p>

**Comment by mbledkowski** [comment]
Why is this feature not supported?

**Reply by RogueScholar** [reply]
I don't understand it either. Back in 2018 when the feature was new, okay, I can see the benefit to waiting for a new feature to be acid tested before rolling it out in production. But now? It's been in use for over six years, and this thread and a similar one at GitLab's gitaly repo are testaments that there are at least a few of us out there using it day-to-day…seems to me like its time has come, especially when the world we live in only seems to get less, not more, secure.

**Comment by LeviPesin** [comment]
This seems to be not very hard to enable, why is it still not?

**Comment by suhasdotcom** [comment]
They may have a good reason to not do it. I do not know what it may be. It seems quiet counter-intuitive to not support it.

If someone wants to support this feature on a remote that they control, check this [article](https://people.kernel.org/monsieuricon/signed-git-pushes).

**Comment by adentinger** [comment]
For those interested, you can tell Git to perform a signed push only when supported by setting `--signed=if-asked`. Or do it by default with a git config:

```bash
git config --global push.gpgSign if-asked
```

The latter _will_ show you a warning whenever you push though:

```
warning: not sending a push certificate since the receiving end does not support --signed push
```

See my git signing setup [here](https://github.com/adentinger/anthonyize/blob/e488132bc6a7cf181413b83f9695fa8084eba11b/scripts/commands#L183) for an example.

**Reply by sergiorgiraldo** [reply]
2026, still not accepted on GH
